doc/writing-specs.markdown

```diff
@@ -158,3 +158,20 @@ The report produced at the end of a test run counts passed, failed, and pending
 spec items separately, and having unresolved pending spec items does not cause
 your test suite to fail; that is, it can still exit with a status code of 0 if
 there are unresolved pending spec items.
+
+### Generating your specs dinamically
+
+You can generate your specs dinamically using the `runIO` function. For
+instance, lets say you want to generate a spec per file in a folder. This
+can be done the following way:
+
+```hspec
+main :: IO ()
+main = hspec $ do
+  describe "generate specs" $ do
+    allFiles <- runIO (getDirectoryContents myDirPath)
+    let files = filter (`notElem` [".", ".."]) allFiles
+    forM_ files $ \file -> do
+      it ("test using the file " ++ file) $ do
+        pending
+```
```

Comment:
Can you try to make this example executable, similar to what we already have under `doc/_includes/`?